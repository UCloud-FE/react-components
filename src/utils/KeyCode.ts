export default {
    ZERO: 48,
    NINE: 57,

    NUMPAD_ZERO: 96,
    NUMPAD_NINE: 105,

    BACKSPACE: 8,
    TAB: 9,
    ENTER: 13,
    DELETE: 46,

    ARROW_UP: 38,
    ARROW_DOWN: 40,
    ARROW_LEFT: 37,
    ARROW_RIGHT: 39
};
