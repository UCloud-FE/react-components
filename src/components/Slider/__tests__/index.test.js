import React from 'react';
import { mount } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';

import Slider from 'src/components/Slider';
import KeyCode from 'src/interfaces/KeyCode';
import sleep from 'tests/shared/sleep';

jest.unmock('rc-trigger');

describe('Slider', () => {
    test('float', () => {
        const onChange = jest.fn();
        let onChangeCalledTimes = 0;
        const wrapper = mount(<Slider min={0} max={100} step={0.1} onChange={onChange} />);
        const input = wrapper.find('input');

        input.simulate('focus');
        input.simulate('change', { target: { value: '4.11111111' } });
        input.simulate('blur');
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onChange).toHaveBeenCalledWith(4.1);
    });
    test('onLastChange', () => {
        const onChange = jest.fn(v => (currentValue = v));
        const onLastChange = jest.fn();
        let onChangeCalledTimes = 0;
        let onLastChangeCalledTimes = 0;
        let currentValue = 12;
        const wrapper = mount(
            <Slider
                min={10}
                max={1000}
                onChange={onChange}
                onLastChange={onLastChange}
                isSensitive
                defaultValue={currentValue}
            />
        );

        const input = wrapper.find('input');

        expect(input.length).toBe(1);
        input.simulate('focus');

        const testChange = shouldTriggerChange => {
            expect(onChange).toHaveBeenCalledTimes(shouldTriggerChange ? ++onChangeCalledTimes : onChangeCalledTimes);
        };
        const changeAndEnter = (v, matchRule, shouldTriggerChange = true) => {
            input.simulate('change', { target: { value: v } });
            testChange(matchRule && shouldTriggerChange);
            input.simulate('keydown', { keyCode: KeyCode['ENTER'] });
            testChange(!matchRule && shouldTriggerChange);
            expect(onLastChange).toHaveBeenCalledTimes(++onLastChangeCalledTimes);
            expect(`input: ${v}, after press enter, value is: ${currentValue}`).toMatchSnapshot();
        };

        changeAndEnter('13.12312312312');
        changeAndEnter('3.12312312312');
        changeAndEnter('sdasdsa1231(*)@!', false, false);
        changeAndEnter('101001010');
        changeAndEnter('999', true);
        changeAndEnter('950', true);
        changeAndEnter('');
        changeAndEnter(' dsadas --', false, false);

        input.simulate('change', { target: { value: '123' } });
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onLastChange).toHaveBeenCalledTimes(onLastChangeCalledTimes);
        input.simulate('change', { target: { value: '111' } });
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onLastChange).toHaveBeenCalledTimes(onLastChangeCalledTimes);
        input.simulate('change', { target: { value: '213' } });
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onLastChange).toHaveBeenCalledTimes(onLastChangeCalledTimes);
        input.simulate('keydown', { keyCode: KeyCode['ENTER'] });
        expect(onChange).toHaveBeenCalledTimes(onChangeCalledTimes);
        expect(onLastChange).toHaveBeenCalledTimes(++onLastChangeCalledTimes);
        input.simulate('keydown', { keyCode: KeyCode['ENTER'] });
        expect(onChange).toHaveBeenCalledTimes(onChangeCalledTimes);
        expect(onLastChange).toHaveBeenCalledTimes(++onLastChangeCalledTimes);
    });
    test('controlled', () => {
        const onChange = jest.fn();
        const wrapper = mount(
            <Slider
                min={10}
                max={1000}
                onChange={onChange}
                value={12}
                marks={{
                    50: {
                        label: '50',
                        step: 2,
                        ratio: 10
                    },
                    100: {
                        label: '100',
                        step: 5,
                        ratio: 10
                    },
                    200: {
                        label: '200',
                        step: 10,
                        ratio: 15
                    },
                    400: {
                        label: '400',
                        step: 20
                    },
                    1000: {
                        label: '1000',
                        step: 50
                    }
                }}
            />
        );

        const sliderHandler = wrapper.find('div.uc-fe-slider-handle');
        const sliderHandlerStyle = sliderHandler.instance().style;
        expect({ 12: sliderHandlerStyle.left }).toMatchSnapshot();
        wrapper.setProps({
            value: 10
        });
        expect({ 10: sliderHandlerStyle.left }).toMatchSnapshot();
        wrapper.setProps({
            value: 200
        });
        expect({ 200: sliderHandlerStyle.left }).toMatchSnapshot();
        wrapper.setProps({
            value: 210
        });
        expect({ 210: sliderHandlerStyle.left }).toMatchSnapshot();
        wrapper.setProps({
            value: 420
        });
        expect({ 420: sliderHandlerStyle.left }).toMatchSnapshot();
        wrapper.setProps({
            value: 1000
        });
        expect({ 1000: sliderHandlerStyle.left }).toMatchSnapshot();

        const slider = wrapper.find('div.uc-fe-slider');
        expect(sliderHandler.length).toBe(1);
        slider.getDOMNode().getBoundingClientRect = () => ({
            bottom: 0,
            height: 0,
            left: 0,
            right: 0,
            top: 0,
            width: 300
        });
        slider.simulate('mousedown', {
            type: 'mousedown',
            pageX: 10,
            button: 0,
            stopPropagation() {},
            preventDefault() {},
            target: slider.getDOMNode()
        });

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenLastCalledWith(24);
        expect({ 1000: sliderHandlerStyle.left }).toMatchSnapshot();
    });
    test('uncontrolled', () => {
        const onChange = jest.fn(v => (currentValue = v));
        let onChangeCalledTimes = 0;
        let currentValue = 12;
        const wrapper = mount(
            <Slider
                min={10}
                max={1000}
                onChange={onChange}
                defaultValue={currentValue}
                marks={{
                    50: {
                        label: '50',
                        step: 2,
                        ratio: 10
                    },
                    100: {
                        label: '100',
                        step: 5,
                        ratio: 10
                    },
                    200: {
                        label: '200',
                        step: 10,
                        ratio: 15
                    },
                    400: {
                        label: '400',
                        step: 20
                    },
                    1000: {
                        label: '1000',
                        step: 50
                    }
                }}
            />
        );
        const sliderHandler = wrapper.find('div.uc-fe-slider-handle');
        const sliderHandlerStyle = sliderHandler.instance().style;
        const slider = wrapper.find('div.uc-fe-slider');
        expect(sliderHandler.length).toBe(1);
        expect({ [currentValue]: sliderHandlerStyle.left }).toMatchSnapshot();
        slider.getDOMNode().getBoundingClientRect = () => ({
            bottom: 0,
            height: 0,
            left: 0,
            right: 0,
            top: 0,
            width: 300
        });
        slider.simulate('mousedown', {
            type: 'mousedown',
            pageX: 10,
            button: 0,
            stopPropagation() {},
            preventDefault() {},
            target: slider.getDOMNode()
        });
        slider.simulate('mouseup');
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onChange).toHaveBeenLastCalledWith(24);
        expect({ [currentValue]: sliderHandlerStyle.left }).toMatchSnapshot();
        slider.simulate('mousedown', {
            type: 'mousedown',
            pageX: 300,
            button: 0,
            stopPropagation() {},
            preventDefault() {},
            target: slider.getDOMNode()
        });
        slider.simulate('mouseup');
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onChange).toHaveBeenLastCalledWith(1000);
        expect({ [currentValue]: sliderHandlerStyle.left }).toMatchSnapshot();
        slider.simulate('mousedown', {
            type: 'mousedown',
            pageX: 60,
            button: 0,
            stopPropagation() {},
            preventDefault() {},
            target: slider.getDOMNode()
        });
        slider.simulate('mouseup');
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect(onChange).toHaveBeenLastCalledWith(100);
        expect({ [currentValue]: sliderHandlerStyle.left }).toMatchSnapshot();
        slider.simulate('mousedown', {
            type: 'mousedown',
            pageX: 10,
            button: 0,
            stopPropagation() {},
            preventDefault() {},
            target: slider.getDOMNode()
        });
        slider.simulate('mouseup');
        expect(onChange).toHaveBeenLastCalledWith(24);
        expect(onChange).toHaveBeenCalledTimes(++onChangeCalledTimes);
        expect({ [currentValue]: sliderHandlerStyle.left }).toMatchSnapshot();
    });
    test('number input', () => {
        const onChange = jest.fn(v => (currentValue = v));
        let onChangeCalledTimes = 0;
        let currentValue = 12;
        const wrapper = mount(
            <Slider
                min={10}
                max={1000}
                onChange={onChange}
                defaultValue={currentValue}
                marks={{
                    50: {
                        label: '50',
                        step: 2,
                        ratio: 10
                    },
                    100: {
                        label: '100',
                        step: 5,
                        ratio: 10
                    },
                    200: {
                        label: '200',
                        step: 10,
                        ratio: 15
                    },
                    400: {
                        label: '400',
                        step: 20
                    },
                    1000: {
                        label: '1000',
                        step: 50
                    }
                }}
            />
        );

        const input = wrapper.find('input');

        expect(input.length).toBe(1);
        input.simulate('focus');

        const changeAndEnter = (v, shouldTriggerChange = true) => {
            input.simulate('change', { target: { value: v } });
            expect(onChange).toHaveBeenCalledTimes(onChangeCalledTimes);
            input.simulate('keydown', { keyCode: KeyCode['ENTER'] });
            expect(onChange).toHaveBeenCalledTimes(shouldTriggerChange ? ++onChangeCalledTimes : onChangeCalledTimes);
            expect(`input: ${v}, after press enter, value is: ${currentValue}`).toMatchSnapshot();
        };

        changeAndEnter('123');
        changeAndEnter('124', false);
        changeAndEnter('1001');
        changeAndEnter('123');
        changeAndEnter('12');
        changeAndEnter('44');
        changeAndEnter('75');
        changeAndEnter('280');
        changeAndEnter('1000');
        changeAndEnter('asdsaad');
        changeAndEnter('H#OI!*&', false);
        changeAndEnter('123');
        changeAndEnter('');
        changeAndEnter('123');
        changeAndEnter('   dsada   ');
        changeAndEnter('14.12312312312');
        changeAndEnter('14.12312312312', false);

        wrapper.setProps({
            marks: {}
        });

        changeAndEnter('13.12312312312');
        changeAndEnter('3.12312312312');
        changeAndEnter('sdasdsa1231(*)@!', false);
        changeAndEnter('101001010');
        changeAndEnter('999');
        changeAndEnter('950');
        changeAndEnter('');
        changeAndEnter(' dsadas --', false);
    });
    test('isSensitive', () => {
        const onChange = jest.fn(v => (currentValue = v));
        let onChangeCalledTimes = 0;
        let currentValue = 12;
        const wrapper = mount(
            <Slider
                min={10}
                max={1000}
                onChange={onChange}
                isSensitive
                defaultValue={currentValue}
                marks={{
                    50: {
                        label: '50',
                        step: 2,
                        ratio: 10
                    },
                    100: {
                        label: '100',
                        step: 5,
                        ratio: 10
                    },
                    200: {
                        label: '200',
                        step: 10,
                        ratio: 15
                    },
                    400: {
                        label: '400',
                        step: 20
                    },
                    1000: {
                        label: '1000',
                        step: 50
                    }
                }}
            />
        );

        const input = wrapper.find('input');

        expect(input.length).toBe(1);
        input.simulate('focus');

        const testChange = shouldTriggerChange => {
            expect(onChange).toHaveBeenCalledTimes(shouldTriggerChange ? ++onChangeCalledTimes : onChangeCalledTimes);
        };
        const changeAndEnter = (v, matchRule, shouldTriggerChange = true) => {
            input.simulate('change', { target: { value: v } });
            testChange(matchRule && shouldTriggerChange);
            input.simulate('keydown', { keyCode: KeyCode['ENTER'] });
            testChange(!matchRule && shouldTriggerChange);
            expect(`input: ${v}, after press enter, value is: ${currentValue}`).toMatchSnapshot();
        };

        changeAndEnter('120', true);
        changeAndEnter('124', false, false);
        changeAndEnter('1001');
        changeAndEnter('123');
        changeAndEnter('12', true);
        changeAndEnter('44', true);
        changeAndEnter('75', true);
        changeAndEnter('280', true);
        changeAndEnter('1000', true);
        changeAndEnter('asdsaad', false, true);
        changeAndEnter('H#OI!*&', false, false);
        changeAndEnter('123');
        changeAndEnter('');
        changeAndEnter('123');
        changeAndEnter('   dsada   ');
        changeAndEnter('14.12312312312');
        changeAndEnter('14.12312312312', false, false);

        wrapper.setProps({
            marks: {}
        });

        changeAndEnter('13.12312312312');
        changeAndEnter('3.12312312312');
        changeAndEnter('sdasdsa1231(*)@!', false, false);
        changeAndEnter('101001010');
        changeAndEnter('999', true);
        changeAndEnter('950', true);
        changeAndEnter('');
        changeAndEnter(' dsadas --', false, false);
    });
    test('numberInputTipFormatter', async () => {
        const wrapper = mount(<Slider min={10} defaultValue={10} max={1000} />);

        const input = wrapper.find('input');

        expect(input.length).toBe(1);
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
        input.simulate('focus');
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
        wrapper.setProps({
            numberInputTipFormatter: null
        });
        await sleep(500);
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
        wrapper.setProps({
            numberInputTipFormatter: function tip(option) {
                const { currentValue, inputValue, isSensitive } = option;
                return (
                    <span>
                        currentValue: {currentValue}
                        inputValue: {inputValue}
                        isSensitive: {isSensitive + ''}
                    </span>
                );
            }
        });
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
        input.simulate('change', { target: { value: 'asdasd' } });
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
        wrapper.setProps({
            isSensitive: true
        });
        expect(renderToJson(wrapper.render())).toMatchSnapshot();
    });
});
