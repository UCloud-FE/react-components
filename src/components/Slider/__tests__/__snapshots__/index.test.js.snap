// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider controlled 1`] = `
Object {
  "12": "0.6666666666666667%",
}
`;

exports[`Slider controlled 2`] = `
Object {
  "10": "0%",
}
`;

exports[`Slider controlled 3`] = `
Object {
  "200": "35%",
}
`;

exports[`Slider controlled 4`] = `
Object {
  "210": "35.66666666666667%",
}
`;

exports[`Slider controlled 5`] = `
Object {
  "420": "53%",
}
`;

exports[`Slider controlled 6`] = `
Object {
  "1000": "100%",
}
`;

exports[`Slider controlled 7`] = `
Object {
  "1000": "100%",
}
`;

exports[`Slider isSensitive 1`] = `"input: 120, after press enter, value is: 120"`;

exports[`Slider isSensitive 2`] = `"input: 124, after press enter, value is: 120"`;

exports[`Slider isSensitive 3`] = `"input: 1001, after press enter, value is: 1000"`;

exports[`Slider isSensitive 4`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider isSensitive 5`] = `"input: 12, after press enter, value is: 12"`;

exports[`Slider isSensitive 6`] = `"input: 44, after press enter, value is: 44"`;

exports[`Slider isSensitive 7`] = `"input: 75, after press enter, value is: 75"`;

exports[`Slider isSensitive 8`] = `"input: 280, after press enter, value is: 280"`;

exports[`Slider isSensitive 9`] = `"input: 1000, after press enter, value is: 1000"`;

exports[`Slider isSensitive 10`] = `"input: asdsaad, after press enter, value is: 10"`;

exports[`Slider isSensitive 11`] = `"input: H#OI!*&, after press enter, value is: 10"`;

exports[`Slider isSensitive 12`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider isSensitive 13`] = `"input: , after press enter, value is: 10"`;

exports[`Slider isSensitive 14`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider isSensitive 15`] = `"input:    dsada   , after press enter, value is: 10"`;

exports[`Slider isSensitive 16`] = `"input: 14.12312312312, after press enter, value is: 14"`;

exports[`Slider isSensitive 17`] = `"input: 14.12312312312, after press enter, value is: 14"`;

exports[`Slider isSensitive 18`] = `"input: 13.12312312312, after press enter, value is: 13"`;

exports[`Slider isSensitive 19`] = `"input: 3.12312312312, after press enter, value is: 10"`;

exports[`Slider isSensitive 20`] = `"input: sdasdsa1231(*)@!, after press enter, value is: 10"`;

exports[`Slider isSensitive 21`] = `"input: 101001010, after press enter, value is: 1000"`;

exports[`Slider isSensitive 22`] = `"input: 999, after press enter, value is: 999"`;

exports[`Slider isSensitive 23`] = `"input: 950, after press enter, value is: 950"`;

exports[`Slider isSensitive 24`] = `"input: , after press enter, value is: 10"`;

exports[`Slider isSensitive 25`] = `"input:  dsadas --, after press enter, value is: 10"`;

exports[`Slider number input 1`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider number input 2`] = `"input: 124, after press enter, value is: 120"`;

exports[`Slider number input 3`] = `"input: 1001, after press enter, value is: 1000"`;

exports[`Slider number input 4`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider number input 5`] = `"input: 12, after press enter, value is: 12"`;

exports[`Slider number input 6`] = `"input: 44, after press enter, value is: 44"`;

exports[`Slider number input 7`] = `"input: 75, after press enter, value is: 75"`;

exports[`Slider number input 8`] = `"input: 280, after press enter, value is: 280"`;

exports[`Slider number input 9`] = `"input: 1000, after press enter, value is: 1000"`;

exports[`Slider number input 10`] = `"input: asdsaad, after press enter, value is: 10"`;

exports[`Slider number input 11`] = `"input: H#OI!*&, after press enter, value is: 10"`;

exports[`Slider number input 12`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider number input 13`] = `"input: , after press enter, value is: 10"`;

exports[`Slider number input 14`] = `"input: 123, after press enter, value is: 120"`;

exports[`Slider number input 15`] = `"input:    dsada   , after press enter, value is: 10"`;

exports[`Slider number input 16`] = `"input: 14.12312312312, after press enter, value is: 14"`;

exports[`Slider number input 17`] = `"input: 14.12312312312, after press enter, value is: 14"`;

exports[`Slider number input 18`] = `"input: 13.12312312312, after press enter, value is: 13"`;

exports[`Slider number input 19`] = `"input: 3.12312312312, after press enter, value is: 10"`;

exports[`Slider number input 20`] = `"input: sdasdsa1231(*)@!, after press enter, value is: 10"`;

exports[`Slider number input 21`] = `"input: 101001010, after press enter, value is: 1000"`;

exports[`Slider number input 22`] = `"input: 999, after press enter, value is: 999"`;

exports[`Slider number input 23`] = `"input: 950, after press enter, value is: 950"`;

exports[`Slider number input 24`] = `"input: , after press enter, value is: 10"`;

exports[`Slider number input 25`] = `"input:  dsadas --, after press enter, value is: 10"`;

exports[`Slider numberInputTipFormatter 1`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW eTIEYH"
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx cEshfv"
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="10"
      />
    </div>
  </div>
</div>
`;

exports[`Slider numberInputTipFormatter 2`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW cYGvEs"
    style=""
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx cEshfv"
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="10"
      />
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="uc-fe-popover style__PopoverWrap-cfqvNq iMAKfT uc-fe-slider-tooltip uc-fe-popover-placement-top uc-fe-popover-animation-fade-appear"
        style="left: -999px; top: -1006px;"
      >
        <div
          class="style__TooltipWrap-gUIDEH eERSIJ"
        >
          <span
            class="style__Arrow-bfHBbq jAjcsO"
          >
            <span
              class="style__ArrowInner-cpbAEO jbBSfh"
            />
          </span>
          <div
            class="style__ContentWrap-kpQkHV gMGmx"
          >
            当前生效值为10，输入值为10
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider numberInputTipFormatter 3`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW cYGvEs"
    style=""
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx cEshfv"
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="10"
      />
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="uc-fe-popover style__PopoverWrap-cfqvNq iMAKfT uc-fe-slider-tooltip uc-fe-popover-placement-top uc-fe-popover-hidden"
        style="left: -999px; top: -1006px;"
      >
        <div
          class="style__TooltipWrap-gUIDEH eERSIJ"
        >
          <span
            class="style__Arrow-bfHBbq jAjcsO"
          >
            <span
              class="style__ArrowInner-cpbAEO jbBSfh"
            />
          </span>
          <div
            class="style__ContentWrap-kpQkHV gMGmx"
          >
            当前生效值为10，输入值为10
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider numberInputTipFormatter 4`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW cYGvEs"
    style=""
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx cEshfv"
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="10"
      />
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="uc-fe-popover style__PopoverWrap-cfqvNq iMAKfT uc-fe-slider-tooltip uc-fe-popover-placement-top uc-fe-popover-animation-fade-enter"
        style="left: -999px; top: -1006px;"
      >
        <div
          class="style__TooltipWrap-gUIDEH eERSIJ"
        >
          <span
            class="style__Arrow-bfHBbq jAjcsO"
          >
            <span
              class="style__ArrowInner-cpbAEO jbBSfh"
            />
          </span>
          <div
            class="style__ContentWrap-kpQkHV gMGmx"
          >
            <span>
              currentValue: 
              10
              inputValue: 
              10
              isSensitive: 
              undefined
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider numberInputTipFormatter 5`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW cYGvEs"
    style=""
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx dXwOlJ"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="asdasd"
      />
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="uc-fe-popover style__PopoverWrap-cfqvNq iMAKfT uc-fe-slider-tooltip uc-fe-popover-placement-top"
        style="left: -999px; top: -1006px;"
      >
        <div
          class="style__TooltipWrap-gUIDEH eERSIJ"
        >
          <span
            class="style__Arrow-bfHBbq jAjcsO"
          >
            <span
              class="style__ArrowInner-cpbAEO jbBSfh"
            />
          </span>
          <div
            class="style__ContentWrap-kpQkHV gMGmx"
          >
            <span>
              currentValue: 
              10
              inputValue: 
              asdasd
              isSensitive: 
              undefined
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider numberInputTipFormatter 6`] = `
<div
  class="style__SliderWrap-caqjLj hMffQf"
>
  <div
    class="uc-fe-slider"
  >
    <div
      class="uc-fe-slider-rail"
    />
    <div
      class="uc-fe-slider-track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="uc-fe-slider-step"
    />
    <div
      aria-disabled="false"
      aria-valuemax="300"
      aria-valuemin="0"
      aria-valuenow="10"
      class="uc-fe-slider-handle"
      role="slider"
      style="left: 0%;"
      tabindex="0"
    />
    <div
      class="uc-fe-slider-mark"
    />
  </div>
  <div
    class="style__NumberInputWrap-cGKKuW cYGvEs"
    style=""
  >
    <div>
      <span
        class="style__HandlerUp-laNuFx dXwOlJ"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-up style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
      <span
        class="style__HandlerDown-gQSTYw guDwuU"
        disabled=""
        unselectable="unselectable"
      >
        <i
          class="icon__caret-down style__IconWrap-hmdRqY gdfCrX"
        />
      </span>
    </div>
    <div
      class="style__InputWrap-fzLmJu fPsDRG"
    >
      <input
        autocomplete="off"
        class="style__Input-dtDjMm esRyZC"
        max="1000"
        min="10"
        step="1"
        value="asdasd"
      />
    </div>
  </div>
  <div
    style="position: absolute; top: 0px; left: 0px; width: 100%;"
  >
    <div>
      <div
        class="uc-fe-popover style__PopoverWrap-cfqvNq iMAKfT uc-fe-slider-tooltip uc-fe-popover-placement-top"
        style="left: -999px; top: -1006px;"
      >
        <div
          class="style__TooltipWrap-gUIDEH eERSIJ"
        >
          <span
            class="style__Arrow-bfHBbq jAjcsO"
          >
            <span
              class="style__ArrowInner-cpbAEO jbBSfh"
            />
          </span>
          <div
            class="style__ContentWrap-kpQkHV gMGmx"
          >
            <span>
              currentValue: 
              10
              inputValue: 
              asdasd
              isSensitive: 
              true
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider onLastChange 1`] = `"input: 13.12312312312, after press enter, value is: 13"`;

exports[`Slider onLastChange 2`] = `"input: 3.12312312312, after press enter, value is: 10"`;

exports[`Slider onLastChange 3`] = `"input: sdasdsa1231(*)@!, after press enter, value is: 10"`;

exports[`Slider onLastChange 4`] = `"input: 101001010, after press enter, value is: 1000"`;

exports[`Slider onLastChange 5`] = `"input: 999, after press enter, value is: 999"`;

exports[`Slider onLastChange 6`] = `"input: 950, after press enter, value is: 950"`;

exports[`Slider onLastChange 7`] = `"input: , after press enter, value is: 10"`;

exports[`Slider onLastChange 8`] = `"input:  dsadas --, after press enter, value is: 10"`;

exports[`Slider uncontrolled 1`] = `
Object {
  "12": "0.6666666666666667%",
}
`;

exports[`Slider uncontrolled 2`] = `
Object {
  "24": "3.6666666666666665%",
}
`;

exports[`Slider uncontrolled 3`] = `
Object {
  "1000": "100%",
}
`;

exports[`Slider uncontrolled 4`] = `
Object {
  "100": "20%",
}
`;

exports[`Slider uncontrolled 5`] = `
Object {
  "24": "3.6666666666666665%",
}
`;
